packages/evm/test/treasury/protocol-treasury.js

```diff
@@ -162,194 +164,224 @@ contract('ProtocolTreasury', ([_, disputeManager, holder, someone]) => {
-              assertBn(balance, amount, 'token balance do not match')
-            })
+                const balance = await DAI.balanceOf(recipient)
+                assertBn(balance, amount, 'token balance do not match')
+              })
 
-            it('emits an event', async () => {
-              const receipt = await treasury.withdraw(DAI.address, recipient, amount, { from })
+              it('emits an event', async () => {
+                const receipt = await withdraw(DAI, recipient, amount, sender, authorize)
 
-              assertAmountOfEvents(receipt, TREASURY_EVENTS.WITHDRAW)
-              assertEvent(receipt, TREASURY_EVENTS.WITHDRAW, { expectedArgs: { from, to: recipient, token: DAI.address, amount } })
-            })
+                assertAmountOfEvents(receipt, TREASURY_EVENTS.WITHDRAW)
+                assertEvent(receipt, TREASURY_EVENTS.WITHDRAW, { expectedArgs: { from: holder, to: recipient, token: DAI, amount } })
+              })
 
-            it('does not affect other token balances', async () => {
-              const previousANTBalance = await treasury.balanceOf(ANT.address, recipient)
+              it('does not affect other token balances', async () => {
+                const previousANTBalance = await treasury.balanceOf(ANT.address, recipient)
 
-              await treasury.withdraw(DAI.address, recipient, amount, { from })
+                await withdraw(DAI, recipient, amount, sender, authorize)
 
-              const currentANTBalance = await treasury.balanceOf(ANT.address, recipient)
-              assertBn(currentANTBalance, previousANTBalance, 'account balance do not match')
+                const currentANTBalance = await treasury.balanceOf(ANT.address, recipient)
+                assertBn(currentANTBalance, previousANTBalance, 'account balance do not match')
+              })
+            })
+
+            context('when the treasury contract does not have enough tokens', () => {
+              it('reverts', async () => {
+                await assertRevert(withdraw(DAI, recipient, amount, sender, authorize), TREASURY_ERRORS.WITHDRAW_FAILED)
+              })
             })
           })
 
-          context('when the treasury contract does not have enough tokens', () => {
+          context('when the given amount is grater than the balance of the account', () => {
+            const amount = bigExp(201, 18)
+
             it('reverts', async () => {
-              await assertRevert(treasury.withdraw(DAI.address, recipient, amount, { from }), TREASURY_ERRORS.WITHDRAW_FAILED)
+              await assertRevert(withdraw(DAI, recipient, amount, sender, authorize), TREASURY_ERRORS.WITHDRAW_INVALID_AMOUNT)
             })
           })
         })
 
-        context('when the given amount is grater than the balance of the account', () => {
-          const amount = bigExp(201, 18)
+        context('when the given recipient is the zero address', () => {
```

Comment:
I found this test kind of weird, do we have any explicit checks on the zero address? If so, should we also check that tokens are not being sent back to the treasury contract?